src/dynarmic/backend/riscv64/emit_riscv64.cpp

```diff
@@ -5,25 +5,97 @@
 
 #include "dynarmic/backend/riscv64/emit_riscv64.h"
 
+#include <bit>
+
 #include <biscuit/assembler.hpp>
+#include <fmt/ostream.h>
 
 #include "dynarmic/backend/riscv64/a32_jitstate.h"
+#include "dynarmic/backend/riscv64/abi.h"
+#include "dynarmic/backend/riscv64/emit_context.h"
+#include "dynarmic/backend/riscv64/reg_alloc.h"
 #include "dynarmic/ir/basic_block.h"
+#include "dynarmic/ir/microinstruction.h"
+#include "dynarmic/ir/opcodes.h"
 
 namespace Dynarmic::Backend::RV64 {
 
-EmittedBlockInfo EmitRV64(biscuit::Assembler& as, [[maybe_unused]] IR::Block block) {
+// TODO: We should really move this to biscuit.
+static void Mov64(biscuit::Assembler& as, biscuit::GPR rd, s64 imm) {
+    if (((imm << 32) >> 32) == imm) {
```

Comment:
```suggestion
    if (mcl::bit::sign_extend<32>(imm) == imm) {
```

`#include <mcl/bit/bit_field.hpp>`